How to run the examples

I would like to test the examples but I am running into some problems.

On Linux with a NVIDA 1060 I get this error:
```
➜  vulkano-examples git:(master) cargo run --bin triangle
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `target/debug/triangle`
thread 'main' panicked at 'failed to create Vulkan instance: LoadingError(LibraryLoadFailure("libvulkan.so.1: cannot open shared object file: No such file or directory"))', /checkout/src/libcore/result.rs:860:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

Do I need to install something from Vulkan?

On MacOS I get this error:
```
  = note: ld: framework not found MoltenVK
          clang: error: linker command failed with exit code 1 (use -v to see invocation)


error: aborting due to previous error

error: Could not compile `vulkano-examples`.
```

Do I need to install anything on macOS to make it work?


Thank you very much for any help.

I downloaded Molten 0.19.0, dropped `MoltenVK.framework` into `/Library/Framework/` and `libMoltenVK.dylib` into `/usr/local/lib` (just in case ;-) ), but I get following error when executing `cargo run --bin triangle`:

```
 = note: Undefined symbols for architecture x86_64:
            "_kIOSurfaceWidth", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_kIOSurfaceHeight", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_kIOSurfaceElementWidth", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_kIOSurfaceElementHeight", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_kIOSurfaceIsGlobal", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_kIOSurfaceBytesPerElement", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_IOSurfaceGetBytesPerElement", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_IOSurfaceGetElementWidth", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_IOSurfaceGetHeight", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_IOSurfaceCreate", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_IOSurfaceGetWidth", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
            "_IOSurfaceGetElementHeight", referenced from:
                MVKImage::useIOSurface(__IOSurface*) in MoltenVK(MoltenVK-x86_64-master.o)
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Sorry for the noise, I was trying to run example from `volkano-examples` repo. When I run it from this repo dependecies look more fresh, but I get then another error:

```
  Compiling vulkano-win v0.7.3 (file:///Users/rprakapchuk/Hub/vulkano/vulkano-win)
error[E0599]: no variant named `new` found for type `metal::CAMetalLayer` in the current scope
   --> vulkano-win/src/lib.rs:168:17
    |
168 |     let layer = CAMetalLayer::new();
    |                 ^^^^^^^^^^^^^^^^^ variant not found in `metal::CAMetalLayer`

error: aborting due to previous error

error: Could not compile `vulkano-win`.
```

--- a/vulkano/build.rs
+++ b/vulkano/build.rs
@@ -12,6 +12,7 @@ use std::env;
 fn main() {
     let target = env::var("TARGET").unwrap();
     if target.contains("apple-darwin") {
+        println!("cargo:rustc-link-search=framework=/Library/Frameworks/");
         println!("cargo:rustc-link-lib=c++");
         println!("cargo:rustc-link-lib=framework=MoltenVK");
         println!("cargo:rustc-link-lib=framework=IOSurface");

